R/result.R

```diff
@@ -1,36 +1,343 @@
+
+  public = list(
+    initialize = function(name,
+                          result,
+                          params,
+                          tags = NULL,
+                          info = NULL,
+                          context = NULL,
+                          github = NULL,
+                          options = NULL,
+                          output = NULL,
+                          rscript = NULL) {
+      self$name <- name
+      self$result <- result
+      self$params <- params
+      self$tags <- tags
+      self$info <- info
+      self$context <- context
+      self$github <- github
+      self$options <- options
+      self$output <- output
+      self$rscript <- rscript
+    },
+
+    data.frame = function(row.names = NULL, optional = FALSE, packages = "arrow", ...) {
+      x <- self$list
+
+      pkgs <- x$params$packages
+      x$params$packages <- NULL
+      for (p in packages) {
+        x$params[[paste0("version_", p)]] <- pkgs[p, "version"]
+      }
+
+      to_list_col <- lengths(x$params) != 1L
+      x$params[to_list_col] <- lapply(x$params[to_list_col], list)
+
+      out <- dplyr::bind_cols(
+        iteration = seq_len(nrow(x$result)),
+        x$result,
+        dplyr::as_tibble(x$params)
+      )
+      out$output <- x$output
+
+      # append metadata fields to dataframe as attributes
+      metadata_elements <- c("name", "tags", "info", "context", "github", "options")
+      for (element in metadata_elements) {
+        if (element %in% names(x)) {
+          attr(out, element) <- x[[element]]
+        }
+      }
+
+      out
+    }
+  ),
+
+  active = list(
+    name = function(name) {
+      if (!missing(name)) {
+        private$.name <- name
+      }
+      private$.name
+    },
+    result = function(result) {
+      if (!missing(result)) {
+        private$.result <- result
+      }
+      private$.result
+    },
+    params = function(params) {
+      if (!missing(params)) {
+        private$.params <- params
+      }
+      private$.params
+    },
+    tags = function(tags) {
+      if (!missing(tags)) {
+        private$.tags <- tags
+      }
+      private$.tags
+    },
+    info = function(info) {
+      if (!missing(info)) {
+        private$.info <- info
+      }
+      private$.info
+    },
+    context = function(context) {
+      if (!missing(context)) {
+        private$.context <- context
+      }
+      private$.context
+    },
+    github = function(github) {
+      if (!missing(github)) {
+        private$.github <- github
+      }
+      private$.github
+    },
+    options = function(options) {
+      if (!missing(options)) {
+        private$.options <- options
+      }
+      private$.options
+    },
+    output = function(output) {
+      if (!missing(output)) {
+        private$.output <- output
+      }
+      private$.output
+    },
+    rscript = function(rscript) {
+      if (!missing(rscript)) {
+        private$.rscript <- rscript
+      }
+      private$.rscript
+    },
```

Comment:
K, added a `private$get_or_set()` method to make this more compact for now. Left it as a single line for each rather than trying to generate the list; this way it's easy to see what's there as well as change them if/when we want to add validation.